spec/unit/type/cloudwatch_alarm_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'spec_helper'
+
+type_class = Puppet::Type.type(:cloudwatch_alarm)
+
+describe type_class do
+
+  let :params do
+    [
+      :name,
+      :dimensions,
+      :alarm_actions
+    ]
+  end
+
+  let :properties do
+    [
+      :evaluation_periods,
+      :threshold,
+      :comparison_operator,
+      :region,
+      :namespace,
+      :metric,
+      :statistic,
+      :period,
+    ]
+  end
+
+  it 'should have expected properties' do
+    properties.each do |property|
+      expect(type_class.properties.map(&:name)).to be_include(property)
+    end
+  end
```

Comment:
This is not really a useful test, in my opinion. It is much more important to test that the parameter validation works. Example: https://github.com/puppetlabs/puppetlabs-mysql/blob/master/spec/unit/puppet/type/mysql_user_spec.rb